lib/resty/t1k/request.lua

```diff
@@ -299,8 +300,11 @@ function _M.do_request(opts)
     end
 
     local action = t[consts.TAG_HEAD]
+    -- could you replace the comma and question mark with meaningful constants?
     if action == "." then
         ngx.ctx.t1k_request_passed = 1
+        -- could you return the result of t1k check instead of use ngx.ctx.t1k_request_passed and t1k_request_success
+        -- like 0: pass 1: forbidden 2: error, with the error info
```

Comment:
`do_request()` was refactored and result handling was moved to a separate `handler` module, see https://github.com/chaitin/lua-resty-t1k/pull/4/files#diff-91b2a8cb37619d47404014b1d8a7079c70caca0e138098561f2abedc003dd528R1

not only was the result (`action`) needed for result handling, but also for the `status`,  `extra_header`, and `event_id`. 

therefore a `result` table was returned, see https://github.com/chaitin/lua-resty-t1k/pull/4/files#diff-2568e9450c09b870a1cf594a9dc7c5badce59cf4493f425aea4f9295c44b715bR294